### PR TITLE
fix: handle optional deps & prevent long file names in pnpm lockfile v6 used by pnpm 8

### PR DESCRIPTION
**Pull request by gregmagolan**
# Description

Fixes corner cases in v6 lockfile seen downstream. Will add tests cases here in a follow-up when I have more time.

# Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or functionality (non-breaking change which adds functionality)

# Test plan

How has this been tested?

- [x] Local testing

Tested in downstream repository.

### CODE REVIEW

**Review by kormide — APPROVED** [review]

File: npm/private/utils.bzl [hunk 1]
```diff
@@ -106,11 +109,17 @@ def _convert_v6_packages(packages):
...
-        for dep_name, dep_version in deps.items():
-            new_deps[dep_name] = _convert_pnpm_v6_package_name(dep_version)
-        package_info["dependencies"] = new_deps
+        # dependencies
+        dependencies = {}
+        for dep_name, dep_version in package_info.get("dependencies", {}).items():
+            dependencies[dep_name] = _convert_pnpm_v6_package_name(dep_version)
+        package_info["dependencies"] = dependencies
+
+        # optionalDependencies
```
Comment (jbedard) [inline npm/private/utils.bzl:118]: What about dev, peer etc? Are those not in the lockfile? I see `devDependencies` in the method above...

File: npm/private/utils.bzl [hunk 2]
```diff
@@ -66,6 +66,9 @@ def _convert_pnpm_v6_version_peer_dep(version):
         # There is a peer dep if the string ends with ")"
         peer_dep_index = version.find("(")
         peer_dep = version[peer_dep_index:]
+        if len(peer_dep) > 32:
+            # Prevent long paths
```
Comment (jbedard) [inline npm/private/utils.bzl:70]: Is there a reason we could note here?

File: npm/private/utils.bzl (hunk 1 above)
Comment (gregmagolan) [reply npm/private/utils.bzl:118]: They are not in the packages. Resolved dev & peer deps go into dependencies.

File: npm/private/utils.bzl (hunk 2 above)
Comment (jbedard) [reply npm/private/utils.bzl]: Great 👍

**Review by jbedard — APPROVED** [review]
